core/storage/email/gae_models_test.py

```diff
@@ -128,6 +149,168 @@ def test_get_by_hash_behavior_with_sent_datetime_lower_bound(self):
                     sent_datetime_lower_bound='Not a datetime object')
 
 
+class ReplyToIdModelUnitTests(test_utils.GenericTestBase):
+    """Test the GeneralFeedbackEmailReplyToIdModel class."""
+
+    def setUp(self):
+        super(ReplyToIdModelUnitTests, self).setUp()
+
+        def _generate_unique_id_for_tests(unused_cls):
+            return 'Non_unique_id'
+
+        self.generate_constant_id_ctx = self.swap(
+            email_models.GeneralFeedbackEmailReplyToIdModel,
+            '_generate_unique_reply_to_id',
+            types.MethodType(
+                _generate_unique_id_for_tests,
+                email_models.GeneralFeedbackEmailReplyToIdModel))
+
+    def test_id_generation_is_correctly_formatted(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created = model.create('user_id', 'thread_id')
+        self.assertEqual(created.id, 'user_id.thread_id')
+
+        created = model.create('other_user_id', 'other_thread_id')
+        self.assertNotEqual(created.id, 'user_id.thread_id')
+
+    def test_generated_reply_to_id_is_unique(self):
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        all_ids = [
+            model.create('user_id', 'thread_id').reply_to_id
+            for _ in xrange(100)]
+        set_ids = set(all_ids) # Set contains unique values
+        self.assertEqual(len(all_ids), len(set_ids))
+
+    def test_create_raises_when_duplicate_ids(self):
+        with self.generate_constant_id_ctx, self.assertRaises(Exception):
```

Comment:
Can you use assertRaisesRegexp() to include part of the exception message? Otherwise any exception being raised here will cause this test to pass, even if the exception is unrelated to the behavior being tested here.

Reply:
Done ! Didn't knew it existed, it's quite useful !